placeholder-transaction-trait/src/transaction_trait.rs

```diff
@@ -0,0 +1,57 @@
+use {
+    crate::MessageTrait,
+    solana_sdk::{
+        hash::Hash,
+        packet::Packet,
+        signature::Signature,
+        transaction::{TransactionAccountLocks, TransactionError, VersionedTransaction},
+    },
+};
+
+mod reference;
+mod sanitized_transaction;
+
+pub trait TransactionTrait: MessageTrait {
+    /// Get the first signature of the message.
+    fn signature(&self) -> &Signature;
+
+    /// Get all the signatures of the message.
+    fn signatures(&self) -> &[Signature];
+
+    /// Returns the message hash.
+    // TODO: consider moving this to Message
+    fn message_hash(&self) -> &Hash;
+
+    /// Returns true if the transaction is a simple vote transaction.
+    // TODO: consider moving this to Message
+    fn is_simple_vote_transaction(&self) -> bool;
+
+    /// Validate and return the account keys locked by this transaction
+    // TODO: Remove this once all uses are gone. (Allocation inside of this)
+    fn get_account_locks(
```

Comment:
`get_account_locks`, `get_account_locks_unchecked`, `to_versioned_transaction`, `to_packet`.

Seems these could all be removed, and the few uses implemented in some core/accounts internal trait until the uses are removed.

Reply:
removed `get_account_locks`, `get_account_locks_unchecked`, `packet`. They don't need to live on this interface.